src/main/kotlin/com/exactpro/th2/codec/fixng/FixNgCodec.kt

```diff
@@ -150,54 +152,113 @@ class FixNgCodec(dictionary: IDictionaryStructure, settings: FixNgCodecSettings)
         return MessageGroup(messages)
     }
 
-    private fun Field.decode(source: ByteBuf, target: MutableMap<String, Any>, value: String, tag: Int, context: IReportingContext) {
-        val previous = when (this) {
-            is Primitive -> {
-                val decodedValue = when (primitiveType) {
-                    java.lang.String::class.java -> value
-                    java.lang.Character::class.java -> {
-                        check(value.length == 1) { "Wrong value" }
-                        value[0]
+    private fun processDecodeError(value: Any, isDirty: Boolean, context: IReportingContext, errorMessageText: String) = if (isDirty) {
+        context.warning("Dirty mode decoding WARNING: $errorMessageText")
+        value
+    } else {
+        error(errorMessageText)
+    }
+
+    private fun Field.decode(
+        source: ByteBuf,
+        target: MutableMap<String, Any>,
+        value: String,
+        tag: Int,
+        isDirty: Boolean,
+        context: IReportingContext
+    ) {
+        val decodedValue: Any = if (value.isEmpty()) {
```

Comment:
nitpick: usually, when you have `if else if else` it is better to convert the statement into `when`. This makes clearer the blocks that are executed under different conditions